### PR TITLE
feat:  material mapping fixes odd

### PR DESCRIPTION
**Pull request by asalzburger**
This PR fixes a few things in the material mapping and DD4Hep description:
 * ODD has now by-hand-envelopes (fixes navigation issues), copy and repo is updated
 * Pre/Post step of material user action was wrongly calculated 
 * Naming convention is fixed in `HierachySurface...` and `HierarchyVolume...`
 
Still an issue of not being able to map onto the last boundary surfaces:

ODD_tX0_preliminary

ODD_g4_acts_material_position

### CODE REVIEW

File: Core/include/Acts/Material/SurfaceMaterialMapper.hpp [hunk 1]
```diff
@@ -39,6 +39,8 @@ class TrackingGeometry;
 /// @brief selector for finding surface
 struct MaterialSurface {
   bool operator()(const Surface& sf) const {
+    std::cout << " - checking for SurfaceMaterial " << sf.surfaceMaterial()
```
Comment (Corentin-Allaire) [inline Core/include/Acts/Material/SurfaceMaterialMapper.hpp:42]: Did you forgot to remove this cout ?

File: Examples/Detectors/DD4hepDetector/src/TestTracker/ZPlanarTracker_geo.cpp [hunk 2]
```diff
@@ -80,8 +80,10 @@ static Ref_t create_element(Detector& lcdd, xml_h e, SensitiveDetector sens) {
 
     // add Extension for translation to ACTS TrackingGeometry
     Acts::ActsExtension* detlayer = new Acts::ActsExtension("zyx");
-    detlayer->addValue(10. * Acts::UnitConstants::mm, "r", "envelope");
-    detlayer->addValue(10. * Acts::UnitConstants::mm, "z", "envelope");
+    detlayer->addValue(10. * Acts::UnitConstants::mm, "r_min", "envelope");
+    detlayer->addValue(10. * Acts::UnitConstants::mm, "r_mx", "envelope");
```
Comment (Corentin-Allaire) [inline Examples/Detectors/DD4hepDetector/src/TestTracker/ZPlanarTracker_geo.cpp:84]: Should be `r_max` instead of `r_mx`.

**Review by Corentin-Allaire — COMMENTED** [review]
I have a few small comments but the rest looks good to me.

File: Core/include/Acts/Material/SurfaceMaterialMapper.hpp (hunk 1 above)
Comment (asalzburger) [reply Core/include/Acts/Material/SurfaceMaterialMapper.hpp:42]: yes I did ...

File: Examples/Detectors/DD4hepDetector/src/TestTracker/ZPlanarTracker_geo.cpp (hunk 2 above)
Comment (asalzburger) [reply Examples/Detectors/DD4hepDetector/src/TestTracker/ZPlanarTracker_geo.cpp:84]: Good catch.

**Review by Corentin-Allaire — APPROVED** [review]
Thanks for the fixes. Now it looks good to me !

**Pull request merged**